cobalt/loader/image/animated_webp_image.cc

```diff
@@ -170,36 +167,20 @@ void AnimatedWebPImage::StopInternal() {
 
 void AnimatedWebPImage::StartDecoding() {
   TRACE_EVENT0("cobalt::loader::image", "AnimatedWebPImage::StartDecoding()");
-  lock_.AssertAcquired();
+  DCHECK(task_runner_->RunsTasksInCurrentSequence());
   current_frame_time_ = base::TimeTicks::Now();
-  if (task_runner_->BelongsToCurrentThread()) {
-    DecodeFrames();
-  } else {
-    task_runner_->PostTask(FROM_HERE,
```

Comment:
Why we don't need to PostTask here?

Reply:
Because there's no more locking needed, it's now guaranteed to run in the sequence